src/us/kbase/workspace/test/controllers/sample/SampleServiceController.java

```diff
@@ -0,0 +1,229 @@
+				"gunicorn",
+				"--worker-class", "gevent",
+				"--timeout", "30",
+				"--workers", "1",
+				"--bind", ":" + port,
+				"--log-level", "info",
+				String.format("%s.%sServer:application", SAMPLE_SERVICE_NAME, SAMPLE_SERVICE_NAME))
+				.redirectErrorStream(true)
+				.redirectOutput(logfile.toFile());
+		Map<String, String> env = samplepb.environment();
+		env.put("KB_DEPLOYMENT_CONFIG", ssIniFile.getAbsolutePath().toString());
+		env.put("KB_SERVICE_NAME", SAMPLE_SERVICE_NAME);
+		env.put("PYTHONPATH", libDirPath);
+		samplepb.directory(new File(libDirPath));
+		sampleService = samplepb.start();
+
+		Thread.sleep(1000); //let the service start up
+	}
+
+	private File createSampleServiceDeployCfg(
+			final ArangoController arango,
+			final URL authURL,
+			final String authToken,
+			final URL workspaceURL,
+			final String workspaceToken,
+			final String sampleFullAdminRole,
+			final SampleServiceArangoParameters arangoParams)
+			throws IOException {
+		final File iniFile = tempDir.resolve("sampleService.cfg").toFile();
+		final Ini ini = new Ini();
+		final Section ss = ini.add(SAMPLE_SERVICE_NAME);
+		URL authServiceURL = new URL(authURL.toString() + "/api/legacy/KBase/Sessions/Login");
+		ss.add("auth-service-url", authServiceURL.toString());
+		ss.add("auth-root-url", authURL.toString());
+		ss.add("auth-service-url-allow-insecure", "true");
+		ss.add("auth-token", authToken);
+		ss.add("auth-full-admin-roles", sampleFullAdminRole);
+		
+		ss.add("workspace-url", workspaceURL.toString());
+		ss.add("workspace-read-admin-token", workspaceToken);
+		
+		ss.add("arango-url", "http://localhost:" + arango.getServerPort());
+		ss.add("arango-db", arangoParams.arangoDB);
+		ss.add("arango-user", arangoParams.arangoUser);
+		ss.add("arango-pwd", arangoParams.arangoPwd);
+
+		ss.add("sample-collection", arangoParams.sampleCollection);
+		ss.add("version-collection", arangoParams.versionCollection);
+		ss.add("version-edge-collection", arangoParams.versionEdgeCollection);
+		ss.add("node-collection", arangoParams.nodeCollection);
+		ss.add("node-edge-collection", arangoParams.nodeEdgeCollection);
+		ss.add("data-link-collection", arangoParams.dataLinkCollection);
+		ss.add("workspace-object-version-shadow-collection",
+				arangoParams.workspaceObjectVersionShadowCollection);
+		ss.add("schema-collection", arangoParams.schemaCollection);
+		
+		ini.store(iniFile);
+		return iniFile;
+	}
+
+
+	public int getPort() {
+		return port;
+	}
+
+	public Path getTempDir() {
+		return tempDir;
+	}
+
+	public void destroy(final boolean deleteTempFiles) throws IOException {
+		destroy(deleteTempFiles, false);
+	}
+	
+	public void destroy(final boolean deleteTempFiles, final boolean dumpLogToStdOut)
+			throws IOException {
+		if (sampleService != null) {
+			sampleService.destroy();
+		}
+		if (dumpLogToStdOut) {
+			try (final BufferedReader is = Files.newBufferedReader(logfile)) {
+				is.lines().forEach(l -> System.out.println(l));
+			}
```

Comment:
I'm a little out of the loop with modern Java-isms. Is a `try` without a `catch` allowed these days, and just propagates the uncaught error out to the next layer in the call stack? Is the try really needed here since you're just throwing `IOException`s anyway?

Reply:
`try` without a `catch` is the equivalent of python `with`. In this case it ensures the `BufferedReader is` is closed when the `try` block exits.

Reply:
Or more accurately, `try` with parenthesis around a closeable resource is the equivalent of python `with`. Whether there's a `catch` or not is immaterial.

Reply:
https://docs.oracle.com/javase/8/docs/api/java/lang/AutoCloseable.html

Reply:
Gotcha. LGTM.